beater/server.go

```diff
@@ -225,9 +207,13 @@ func sendStatus(w http.ResponseWriter, r *http.Request, code int, err error) {
 	w.WriteHeader(code)
 
 	if err == nil {
+		responseValid.Inc()
+		logp.Debug("request", "request successful, code=%d", code)
 		return
 	}
 
+	logp.Err("%s, code=%d", err.Error(), code)
```

Comment:
Since at this point I know `err` is not `nil`, semantically makes sense to log it with `logp.Err`, whatever it is.
This reverts a change @ruflin did a while ago, which checked if `code` is `4XX`, and in that case logs it as Info.

Argument IIRC was to not flood log with too much stuff. I don't think that is a concern, but I was ok with that at the time. 
Now I suggest to change it back mostly to keep the code simpler and afford an `if` branch that doesn't give any benefit. Semantically, a `4XX` is still an error, even if the culprit is not the server.

Reply:
From an operator perspective I think an error means there is action needed to "fix" the apm-server. But return a 400 error can happen pretty often especially when exposed to the internet without requiring an action from the operator. I would still keep the old logging.

Reply:
By experience, i can tell you it does *not happen often. 

And if it happens, it usually means that there is a bug in a client, so it needs fixing as well.

Note that agent owners don't have the chance to collect errors from the agent, they need either to get it reported from the end user or from the apm server operator, so is nice if we can highlight it on this side, don't you think?

Reply:
TBH I didn't run apm-server in production yet so that is probably the only way to tell us :-) Both works for me. We are on the same page that if the client sends data which leads to a 404 we should log it as error. Let's go with error and see how it works out in production 👍